### PR TITLE
[Tagger] Add a capacity to the Tagger connector

### PR DESCRIPTION
**Pull request by Lhorus6**
### Proposed changes

* Two new attributes available to act based on contained entities of container (e.g. Report)

### Related issues

* https://github.com/OpenCTI-Platform/connectors/issues/3064

### Checklist

- [X] I consider the submitted work as finished
- [ ] I tested the code for its functionality using different use cases
- [X] I added/update the relevant documentation (either on github or on notion)
- [X] Where necessary I refactored code to improve the overall quality

### Further comments

Need to be tested using the two new attributes available:
- objects-type
- objects-name

### CODE REVIEW

**Review by flavienSindou — COMMENTED** [review]
This code contains duplication. Could you consider refactoring it?

**Comment by Lhorus6** [comment]
Hi @flavienSindou,
I made few modifications. Can you please tell me if this is what you expected?
If not, can you give me some guidelines?

File: internal-enrichment/tagger/src/connector.py [hunk 1]
```diff
@@ -41,34 +43,84 @@ def _process_message(self, data: Dict) -> str:
                     flags = load_re_flags(rule)
 
                     for attribute in rule["attributes"]:
-                        attr = enrichment_entity.get(attribute)
+                        if (attribute.lower() == "objects-type") or (
+                            attribute.lower() == "objects-name"
+                        ):
```
Comment (flavienSindou) [inline internal-enrichment/tagger/src/connector.py]:
```suggestion
                        if attribute.lower() in ["objects-type", "objects-name"]:
```

File: internal-enrichment/tagger/README.md [hunk 2]
```diff
@@ -9,18 +9,27 @@ The reason why this connector can be useful in your workflow is due the fact tha
...
 * label: the actual label that you want to associate to the object
 * search: the regular expression to identify the objects that need the label
-* attributes: list of attributes of the object where the search will be applied. Searches can also be done on labels, in this case specifiy the attribute `objectLabel` in the attribute list.
+* attributes: list of attributes of the object where the search will be applied. Searches can also be done on labels, in this case specify the attribute `objectLabel` in the attribute list.
+
+#### Specific attributes
+
+In the context of containers (i.e. if the scope is a report, grouping, or case entity), it is possible to search among the contained entities. Two options are available:
+
+* Search if a type of entity is contained in the container, in this case specify the attribute `objects-type` in the attribute list.
```
Comment (flavienSindou) [inline internal-enrichment/tagger/README.md]:
```suggestion
* Search if a type of entity is contained in the container, in this case specify the attribute `objects-type` in the attribute list. E.g Report is labeled if it contains System objects : `[{"scopes":["Report"],"rules":[{"label":"with_systems","search":"[Ss]ystem","attributes":["objects-type"]]}]`
```

File: internal-enrichment/tagger/README.md [hunk 3]
```diff
@@ -9,18 +9,27 @@ The reason why this connector can be useful in your workflow is due the fact tha
...
+* Search if a specific entity is contained in the container, in this case specify the attribute `objects-name` in the attribute list.
```
Comment (flavienSindou) [inline internal-enrichment/tagger/README.md]:
```suggestion
* Search if a specific entity is contained in the container, in this case specify the attribute `objects-name` in the attribute list. E.g. Report is labeled if it contains CVE from 2016 : `[{"scopes":["Report"],"rules":[{"label":"with_vuln_2016","search":"CVE-2016","attributes":["objects-name"]]}]`
```

**Review by flavienSindou — APPROVED** [review]
Thanks for your work !

I added few remarks but functional tests are ok.

**Pull request merged**